### ISSUE TITLE
Problems working with git. Learning.

### ISSUE THREAD

**Issue by Gawrion** [HuubMons/HuMo-genealogy#71]
Huub recently merged branches and noticed some files from earlier merges are missing. As we both learn git, github i would be nice to do some case study. I want to compare some files form different repos - Huub is suspecting some files are missing..
I have my local repo in VSC wchich origin is my fork. Earlier i have added remote Huub's repo. Ill check what  remotes repos i got:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git branch
* prepare_next_version
```
Ok - my local working branch is prepare_next_version.
Now i will se what remotes i got:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git remote -v
origin  https://github.com/Gawrion/HuMo-genealogy (fetch)
origin  https://github.com/Gawrion/HuMo-genealogy (push)
prepare-huuba   https://github.com/HuubMons/HuMo-genealogy (fetch)
prepare-huuba   https://github.com/HuubMons/HuMo-genealogy (push)
upstream        https://github.com/HuubMons/HuMo-genealogy.git (fetch)
upstream        https://github.com/HuubMons/HuMo-genealogy.git (push)
```
Ok. I got Huub's repo (name prepare-huuba is my custom name defined when adding repo) so now as we know Huub was restoring/adding some branches we want to see all branches in this repo:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git remote show prepare-huuba
* remote prepare-huuba
  Fetch URL: https://github.com/HuubMons/HuMo-genealogy
  Push  URL: https://github.com/HuubMons/HuMo-genealogy
  HEAD branch: master
  Remote branches:
    master                        tracked
    prepare_next_version          tracked
    revert-68-secure_media_folder new (next fetch will store in remotes/prepare-huuba)
    secure_media_folder           new (next fetch will store in remotes/prepare-huuba)
  Local ref configured for 'git push':
    prepare_next_version pushes to prepare_next_version (local out of date)
```
Ok. We see some new branches. Lets update info about this repo:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git remote update
Fetching origin
Fetching upstream
remote: Enumerating objects: 162, done.
remote: Counting objects: 100% (162/162), done.
remote: Compressing objects: 100% (68/68), done.
remote: Total 162 (delta 94), reused 161 (delta 94), pack-reused 0 (from 0)
Receiving objects: 100% (162/162), 207.06 KiB | 2.07 MiB/s, done.
Resolving deltas: 100% (94/94), completed with 51 local objects.
From https://github.com/HuubMons/HuMo-genealogy
 * [new branch]      master               -> upstream/master
 * [new branch]      prepare_next_version -> upstream/prepare_next_version
 * [new branch]      revert-68-secure_media_folder -> upstream/revert-68-secure_media_folder
 * [new branch]      secure_media_folder  -> upstream/secure_media_folder
Fetching prepare-huuba
From https://github.com/HuubMons/HuMo-genealogy
   674b094..a7bebd1  prepare_next_version -> prepare-huuba/prepare_next_version
 * [new branch]      revert-68-secure_media_folder -> prepare-huuba/revert-68-secure_media_folder
 * [new branch]      secure_media_folder  -> prepare-huuba/secure_media_folder
```
Let's check again if new branches are added and tracked:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git remote show prepare-huuba
* remote prepare-huuba
  Fetch URL: https://github.com/HuubMons/HuMo-genealogy
  Push  URL: https://github.com/HuubMons/HuMo-genealogy
  HEAD branch: master
  Remote branches:
    master                        tracked
    prepare_next_version          tracked
    revert-68-secure_media_folder tracked
    secure_media_folder           tracked
  Local ref configured for 'git push':
    prepare_next_version pushes to prepare_next_version (local out of date)   
```
Ok. All tracked. We can try compare branches with difftool:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git difftool prepare-huuba/secure_media_folder

This message is displayed because 'diff.tool' is not configured.
See 'git difftool --tool-help' or 'git help config' for more details.
'git difftool' will now attempt to use one of the following tools:
kompare emerge vimdiff nvimdiff
```
It seems we need some more steps:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git difftool --tool-help
'git difftool --tool=<tool>' may be set to one of the following:
                vimdiff          Use Vim
                vscode           Use Visual Studio Code (requires a graphical session)
(...)
```
Ok. So let's try to use vscode comparing tool:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git difftool prepare-huuba/secure_media_folder --tool=vscode

Viewing (1/117): 'admin/controller/adminStatisticsController.php'
Launch 'vscode' [Y/n]? y

Viewing (2/117): 'admin/controller/backupController.php'
Launch 'vscode' [Y/n]?
```
It works - tool will iterate through all 117 differences. We know trees differ. In a window comparing tool we cane compare all diffs:
![Screenshot_68](https://github.com/user-attachments/assets/6f4372f8-fdb7-4c82-b0a8-8565356e6feb)

Ok. 117 files are to much. Lets compare file we exactly know is crucial for last changes in my fork:
```
PS C:\Users\baarg\Desktop\HuMo-genealogy fork\HuMo-genealogy\server> git difftool prepare-huuba/secure_media_folder .\admin\views\index_admin.php --tool=vscode

Viewing (1/1): 'admin/views/index_admin.php'
Launch 'vscode' [Y/n]? y
```

As we can see we checked that my changes are indeed missing:
![Screenshot_69](https://github.com/user-attachments/assets/527e28ef-99c4-4746-b1df-df9fe7a80392)

What we can do now to merge both repos again, as we see GitHub tells us "there's nothing to compare"? In next posts.

**Comment by HuubMons** [comment]
I'm not 100% certain, but I noticed this:

- I had created a seperate secure_media_folder, so it was easier to test your changes without interruption of the prepare_next version branch. 

- But: later I noticed you did your changes into the prepare_next_version branch, and then use pull request. I moved the pull request to the secure_media_folder branch.

I did changes in the prepare_next_version branch.
You merged these changes into the secure_media_folder branch.
Then I tried to merge the secure_media_folder into the prepare_next_version branch.

Github doesn't like that. There will be a "there's nothing to compare" message.
For some reason Github doesn't check the changed files or something like that.

Then I reversed some changes and branches. But Github still don't want to merge.

**Comment by Gawrion** [comment]
We're both learning. You can be right about it. But i'm not exactly sure what was the cause. I don't fully understand GitHub logic, but it's better than a week ago :)
As i mentioned on phpBB forum. I forked new fork from p_n_v, cloned it, updated with my changes and filtered out changes that You were missing. I pushed result to my p_n_v branch (literrally i named it that to differ from yours). 

I think that You can securely pull it to prepare_next_version but first analyze it, especially comments i gave to the code. I care about comment so u must understand my logic and corect me if i'm doing someting out of Your coding fhilosophy.

**Comment by HuubMons** [comment]
Ah, perfect!

I just did a pull request to p_n_v, because I don't want to loose your changes again. Thanks for this update!

About coding phylosophy:
I'm trying to introduce the MVC model. But still learning about MVC. This is the reason I had to rename a lot of scripts.
- controller: get data from model, and send to view script.
- model: all PHP code will be moved to the model scripts. If possible no HTML code in the model scripts.
- view: all HTML code will be moved to the view script, if possible no PHP logic in the view scripts (only to show a variable).

At this moment several helper scripts can be found in the include folder. I will probably change multiple functions into classes, so the autoload script will work.

During development, I personally don't mind if PHP code is used in the view script. It will be moved to a model script later.
